File: python/api-gateway/app/logger.py
import json
import sys
from datetime import datetime, timezone

SERVICE_NAME = "api-gateway"


def log(level: str, message: str, **kwargs):
    entry = {
        "timestamp": datetime.now(timezone.utc).isoformat(),
        "level": level,
        "service": SERVICE_NAME,
        "message": message,
        "trace_id": "",
        "span_id": "",
        **kwargs,
    }
    sys.stdout.write(json.dumps(entry) + "\n")
    sys.stdout.flush()

File: nodejs/api-gateway/src/logger.js
'use strict';

const SERVICE_NAME = 'api-gateway';

function log(level, message, extra = {}) {
  const entry = {
    timestamp: new Date().toISOString(),
    level,
    service: SERVICE_NAME,
    message,
    trace_id: '',
    span_id: '',
    ...extra,
  };
  process.stdout.write(JSON.stringify(entry) + '\n');
}

module.exports = { log };
